crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -362,12 +362,11 @@ impl Backend {
                 let mut env = self.env.read().clone();
 
                 let mut node_config = self.node_config.write().await;
-                let (_db, forking) =
+                let (db, forking) =
                     node_config.setup_fork_db(eth_rpc_url, &mut env, &self.fees).await;
 
-                // TODO: Something like this is needed but...
-                // This won't compile because dyn Db is not Sized (and AFAIK cannot be made Sized)
-                // *self.db.write().await = *db.read().await;
+                // TODO: This won't compile because it cannot move out of dereference
+                *self.db.write().await = *db.read().await;
```

Comment:
Specifically I'm not sure how best to really swap the DB here, this doesn't compile.